### PR TITLE
Wait forever for kickstarts on CDROM (#1168902)

### PR DESCRIPTION
**Pull request by wgwoods**
Commit 5d016ef added support for doing cdrom-swapping while doing
"inst.ks=cdrom[:...]". But it turns out that if you're slow to swap the
discs, we'll hit dracut's retry timeout and it'll drop to the emergency
shell.

So, in response to User Requests, we now disable the timeout entirely
when using "inst.ks=cdrom[:...]"; dracut will wait forever for the
kickstart and/or installer media, and if anything goes wrong you're just
stuck and you'll need to reset.

Resolves: rhbz#1168902

### CODE REVIEW

File: dracut/kickstart-genrules.sh [hunk 1]
```diff
@@ -19,6 +19,12 @@ case "${kickstart%%:*}" in
             when_diskdev_appears "$ksdev" \
                 fetch-kickstart-disk \$env{DEVNAME} "$kspath"
         fi
+        # "cdrom:" also means "wait forever for kickstart" because rhbz#1168902
+        if [ "$kstype" = "cdrom" ]; then
+            # if we reset main_loop to 0 every loop, we never hit the timeout..
+            jobfile="$hookdir/initqueue/finished/wait-forever-for-ks-cdrom.sh"
+            echo "main_loop=0" > "$jobfile"
```
Comment (vpodzime) [inline dracut/kickstart-genrules.sh]: I don't now how exactly the ``finished`` guards work, but where does ``main_loop`` come from and if it is a variable, shouldn't the condition be ``$main_loop=0``? Unless I'm missing some big thing here, the comment needs to be extended a bit.

**Comment by wgwoods** [comment]
Modified slightly according to vpodzime's comment.

See [dracut's `dracut-initqueue.sh`](https://github.com/haraldh/dracut/blob/RHEL-7/modules.d/98systemd/dracut-initqueue.sh) for details on how the dracut mainloop works.

**Comment by vpodzime** [comment]
Looks good to me now, thanks.

**Comment by wgwoods** [comment]
Hah whoops, I busted this up. Plus it needs a separate PR for RHEL7, as it turns out, because some stuff never got backported (or got lost in the rebase). So... closing this one.

**Pull request closed without merging**